### PR TITLE
[CPU tests] fix enforce bf16 error

### PR DESCRIPTION
**Pull request by zhaixuejun1993**
### Details:
 - Migrate to 2.0, "InferenceEngine::PluginConfigParams::KEY_ENFORCE_BF16" -> "ov::hint::inference_precision"
 - *...*

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: src/plugins/intel_cpu/tests/functional/shared_tests_instances/core_config.cpp [hunk 1]
```diff
@@ -23,11 +23,6 @@ namespace ov {
 namespace test {
 
 void core_configuration(ov::test::SubgraphBaseTest* test) {
-    #if defined(OPENVINO_ARCH_X86) || defined(OPENVINO_ARCH_X86_64)
-        if (!test->configuration.count(InferenceEngine::PluginConfigParams::KEY_ENFORCE_BF16)) {
-            test->configuration.insert({InferenceEngine::PluginConfigParams::KEY_ENFORCE_BF16, InferenceEngine::PluginConfigParams::NO});
-        }
-    #endif
```
Comment (riverlijunjie) [inline src/plugins/intel_cpu/tests/functional/shared_tests_instances/core_config.cpp:30]: Why remove it?

Comment (zhaixuejun1993) [reply src/plugins/intel_cpu/tests/functional/shared_tests_instances/core_config.cpp:30]: for CPU plugin test, this is no need I think.

Comment (vurusovs) [reply src/plugins/intel_cpu/tests/functional/shared_tests_instances/core_config.cpp:30]: @maxnick please, take a look

Comment (maxnick) [reply src/plugins/intel_cpu/tests/functional/shared_tests_instances/core_config.cpp:30]: `KEY_ENFORCE_BF16` property is not relevant for API 2.0 anymore. But, the check itself still makes sense. We have to force fp32 by default, as on some platfroms tests may be run in bf16 unexpectedly. Thus, if the inference precision is not specified in the config, we enforce fp32. Just like on lines 33-35.

File: src/plugins/intel_cpu/tests/functional/single_layer_tests/split.cpp [hunk 2]
```diff
@@ -54,7 +54,7 @@ class SplitLayerCPUTest : public testing::WithParamInterface<splitCPUTestParams>
         int axis;
         ElementType netPrecision;
         InputShape inputShapes;
-        ov::Shape outIndices;
+        std::vector<size_t> outIndices;
```
Comment (maxnick) [inline src/plugins/intel_cpu/tests/functional/single_layer_tests/split.cpp:57]: Just curious, what is the idea behind changing the type?

**Review by vurusovs — CHANGES_REQUESTED** [review]
please, fix comment about `KEY_ENFORCE_BF16` in terms of API2.0

File: src/plugins/intel_cpu/tests/functional/single_layer_tests/split.cpp (hunk 2 above)
Comment (vurusovs) [reply src/plugins/intel_cpu/tests/functional/single_layer_tests/split.cpp:57]: Previously, it was InferenceEngine::SizeVector, and in most cases it was mean shape, but here `outIndices` is vector of indices, not shape. So it's more about clearness

File: src/plugins/intel_cpu/tests/functional/shared_tests_instances/core_config.cpp (hunk 1 above)
Comment (zhaixuejun1993) [reply src/plugins/intel_cpu/tests/functional/shared_tests_instances/core_config.cpp:30]: @maxnick @vurusovs Pls take a look, modified the if conditional.

**Comment by zhaixuejun1993** [comment]
> please, fix comment about `KEY_ENFORCE_BF16` in terms of API2.0

fixed

**Review by vurusovs — APPROVED** [review]

**Review by riverlijunjie — APPROVED** [review]

File: src/plugins/intel_cpu/tests/functional/shared_tests_instances/core_config.cpp (hunk 1 above)
Comment (zhaixuejun1993) [reply src/plugins/intel_cpu/tests/functional/shared_tests_instances/core_config.cpp:30]: @maxnick do this check & force to f32, will get following test failed:
image
I'm debugging, any suggestions?

Comment (zhaixuejun1993) [reply src/plugins/intel_cpu/tests/functional/shared_tests_instances/core_config.cpp:30]: @maxnick dump the graph & compare, seems here should not force to f32. Pls take a look:
image

**Pull request merged**